packages/headless/src/features/history/history-slice.ts

```diff
@@ -152,6 +155,32 @@ const isCategoryFacetsEqual = (
   return true;
 };
 
+const isAutomaticFacetsEqual = (
+  current: AutomaticFacetSetState,
+  next: AutomaticFacetSetState
+) => {
+  for (const [key, value] of Object.entries(next.set)) {
+    if (!current.set[key]) {
+      return false;
+    }
+
+    const currentSelectedValues = (
+      current.set[key].response.values as FacetValue[]
+    ).filter((value) => value.state === 'selected');
+    const nextSelectedValues = (value.response.values as FacetValue[]).filter(
+      (value) => value.state === 'selected'
+    );
+
+    if (
+      JSON.stringify(currentSelectedValues) !==
```

Comment:
Would they need to be sorted before comparing?

Reply:
I dont think so. Even if I sorted them in some way,  they could never be the same. Doing it this way is fine and is how all the other facets do it. Basically, if the set is different that means a query was done and we to trigger the snapshot action.